droidlet/tools/hitl/nsp_retrain/allocate_instances.py

```diff
@@ -0,0 +1,277 @@
+        print("Registered IP {} at subdomain {}".format(ip, subdomain))
+    except Exception as e:
+        raise e
+
+
+def allocate_instances(instance_num, batch_id, timeout=-1, cf_email="rebeccaqian@fb.com"):
+    instance_ips, instance_ids = request_instance(instance_num, timeout)
+    if os.getenv("CLOUDFLARE_TOKEN") and os.getenv("CLOUDFLARE_ZONE_ID"):
+        logging.info("registering subdomains on craftassist.io")
+        cloudflare_token = os.getenv("CLOUDFLARE_TOKEN")
+        zone_id = os.getenv("CLOUDFLARE_ZONE_ID")
+        cf = CloudFlare.CloudFlare(email=cf_email, token=cloudflare_token)
+        dns_records = cf.zones.dns_records.get(zone_id)
+
+        # Write the subdomains and batch IDs to input CSV for Mephisto
+        # CSV file headers
+        headers = ["subdomain", "batch"]
+        with open("../../../../tools/crowdsourcing/droidlet_static_html_task/data.csv", "w") as fd:
+            csv_writer = csv.writer(fd, delimiter=",")
+            csv_writer.writerow(headers)
+            for x in range(len(instance_ips)):
+                ip = instance_ips[x]
+                subdomain = "dashboard-{}-{}".format(batch_id, x)
+                register_dashboard_subdomain(cf, zone_id, ip, subdomain)
+                # Write record to Mephisto task input CSV
+                csv_writer.writerow([subdomain, batch_id])
+    return instance_ips, instance_ids
+
+
+def free_ecs_instances(instance_ids):
+    for instance_id in instance_ids:
+        try:
+            ecs.stop_task(
+                cluster="craftassist",
+                task=instance_id,
+            )
+        except:
+            pass
+    
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--instance_num", type=int, default=1, help="number of instances requested"
+    )
+    parser.add_argument(
+        "--batch_id",
+        type=int,
+        default=0,
+        help="ID of the current batch, used to track which group of runs the task was run in",
+    )
+    parser.add_argument(
+        "--user", type=str, default="rebeccaqian@fb.com", help="Email of the CloudFlare account"
+    )
+    args = parser.parse_args()
+    instance_ips, instance_ids = request_instance(args.instance_num)
+    batch_id = args.batch_id
+    # register subdomain to proxy instance IP
+    if os.getenv("CLOUDFLARE_TOKEN") and os.getenv("CLOUDFLARE_ZONE_ID"):
+        logging.info("registering subdomains on craftassist.io")
+        cloudflare_token = os.getenv("CLOUDFLARE_TOKEN")
+        zone_id = os.getenv("CLOUDFLARE_ZONE_ID")
+        cf = CloudFlare.CloudFlare(email=args.user, token=cloudflare_token)
+        dns_records = cf.zones.dns_records.get(zone_id)
+
+        # Write the subdomains and batch IDs to input CSV for Mephisto
+        # CSV file headers
+        headers = ["subdomain", "batch"]
+        with open("../../../../tools/crowdsourcing/droidlet_static_html_task/data.csv", "w") as fd:
```

Comment:
A couple of nits here:
1. this looks like a more generic util than only something that `nsp_retrain` can use. Can we rearrange and put this file under `tools/hitl/utils` ?
2. Any reason we are writing to the `fairo/tools/` folder here ? Let's just move `crowdsourcing` to `droidlet/tools` as well

Reply:
> this looks like a more generic util than only something that nsp_retrain can use. Can we rearrange and put this file under tools/hitl/utils ?

thanks for bringing it up. I moved most of reusable files to tools/hitl/utils/.

> Any reason we are writing to the fairo/tools/ folder here ? Let's just move crowdsourcing to droidlet/tools as well

No specific reason -- all the tools are moved to root `util` folder in #688. I've moved crowdsourcing tools back to `droidlet/tools/`